sim-cli/src/parsing.rs

```diff
@@ -259,9 +263,55 @@ async fn validate_activities(
     get_node_info: impl AsyncFn(&PublicKey) -> Result<NodeInfo, LightningError>,
 ) -> Result<Vec<ActivityDefinition>, LightningError> {
     let mut validated_activities = vec![];
+    let mut activity_names = HashSet::new();
 
     // Make all the activities identifiable by PK internally
-    for act in activity.into_iter() {
+    for (index, act) in activity.into_iter().enumerate() {
+        // Generate a default name if one is not provided
+        let name = match &act.name {
+            Some(name) => {
+                // Disallow empty names
+                if name.is_empty() {
+                    return Err(LightningError::ValidationError(
+                        "activity name cannot be empty".to_string()
```

Comment:
nit: let's make this a little more helpful?

"activity name cannot be an empty string, either remove name entirely or provide a string value"